test/lib/QuarkOperationHelper.sol

```diff
@@ -41,17 +44,43 @@ contract QuarkOperationHelper is Test {
                 scriptAddress: scriptAddress,
                 scriptSources: ensureScripts,
                 scriptCalldata: scriptCalldata,
-                nonce: wallet.stateManager().nextNonce(address(wallet)),
+                nonce: semiRandomNonce(wallet),
                 expiry: block.timestamp + 1000
             });
         } else {
             return QuarkWallet.QuarkOperation({
                 scriptAddress: scriptAddress,
                 scriptSources: ensureScripts,
                 scriptCalldata: scriptCalldata,
-                nonce: wallet.stateManager().nextNonce(address(wallet)),
+                nonce: semiRandomNonce(wallet),
                 expiry: block.timestamp + 1000
             });
         }
     }
+
+    /// @dev Note: not sufficiently random for non-test case usage.
+    function semiRandomNonce(QuarkWallet wallet) public view returns (bytes32) {
+        if (address(wallet).code.length == 0) {
+            revert SemiRandomNonceRequiresQuarkStateManagerOrInitializedQuarkWallet(address(wallet));
```

Comment:
I know this is just for test helpers, but I might suggest changing the error to simply `UninitializedQuarkWallet`. Any helper method that relies on the wallet being reified, or I think it's been called "sussed out," can revert with `UninitializedQuarkWallet`. As for knowing that the error came from `semiRandomNonce`, in tests we get a little stack trace, so we should be able to see where the revert originated.